src/translators/typescript.py

```diff
@@ -0,0 +1,748 @@
+
+
+        if isinstance(cond, ast.Is):
+            self._namespace = prev_namespace + ('true_block',)
+            true_branch.accept(self) 
+            self._namespace = prev_namespace + ('false_block',)
+            false_branch.accept(self)
+        else:
+            true_branch.accept(self)
+            false_branch.accept(self)
+
+        self._namespace = prev_namespace
+
+        children_res = self.pop_children_res(children)
+        res = "({}{} ? {} : {})".format(
+            " "*old_ident, children_res[0].strip(),
+            children_res[1].strip(),
+            children_res[2].strip())
+        
+        self.ident = old_ident
+        self._children_res.append(res)
+
+    @append_to
+    def visit_is(self, node):
+        old_ident = self.ident
+        self.ident = 0
+        children = node.children()
+        for c in children:
+            c.accept(self)
+        children_res = self.pop_children_res(children)
+        res = "{}{} {} {}".format(
+            " " * old_ident, children_res[0], str(node.operator),
+            node.rexpr.name)
+        self.ident = old_ident
+        self._children_res.append(res)
+
+    @append_to
+    def visit_new(self, node):
+        old_ident = self.ident
+        self.ident = 0
+        children = node.children()
+        for c in children:
+            c.accept(self)
+        children_res = self.pop_children_res(children)
+        self.ident = old_ident
+        # Remove type arguments from Parameterized Type
+        if getattr(node.class_type, 'can_infer_type_args', None) is True:
+            self._children_res.append("new {}({})".format(
+                " " * self.ident + node.class_type.name,
+                ", ".join(children_res)))
+        else:
+            self._children_res.append("new {}({})".format(
+                " " * self.ident + self.get_type_name(node.class_type),
+                ", ".join(children_res)))
+
+    @append_to
+    def visit_field_access(self, node):
+        old_ident = self.ident
+        self.ident = 0
+        children = node.children()
+        for c in children:
+            c.accept(self)
+        children_res = self.pop_children_res(children)
+        self.ident = old_ident
+        receiver_expr = (children_res[0] if children_res[0]
+                        else "this")
+        res = "{}.{}".format(receiver_expr, node.field)
+        self._children_res.append(res)
+
+    @append_to
+    def visit_func_ref(self, node):
+        def needs_this_prefix(node, decl):
+            if node.receiver is not None:
+                return False
+            elif decl is None:
+                return True # Function is an inherited method
+            elif isinstance(decl, ast.FunctionDeclaration) and decl.is_class_method():
+                return True # Function is method of current class
+            elif isinstance(decl, ast.FieldDeclaration) and decl.get_type().name[:-1] == tst.TypeScriptBuiltinFactory().get_function_type().name[:-1]:
```

Comment:
this line is too long. Please write it across multiple lines or even assign it to a variable.